### ISSUE TITLE
oddness attempting to use a GitHub-provided .zip file as a style package

### ISSUE THREAD

**Issue by wesley-dean-flexion** [vale-cli/vale#585]
tl;dr: specifying a URL to an automatically-generated, GitHub-provided .zip file of a style package won't download and/or extract in a way that the style package can be used by Vale running in the context of a Docker container.  I was able to work around this by extending the jdkato/vale image to download, extract, mangle, and create a new .zip that Vale could use.

It would be nice, however, if another style package source -- a URL ending with `.git` -- could be used that would automagically clone the specified repository into a location that Vale could use.

Proviso: this may or may not be an actual bug; if it is, it may or may not be relate to the Docker image.

I stumbled upon a repository of an individual attempting to codify some of the guidance from PlainLanguage.gov as a Vale style: https://github.com/testthedocs/PlainLanguage

I attempted to use a `.vale.ini` file that that pulled in a .zip file from the master branch of the aforementioned repo like this:

```ini
Packages = Readability, alex, proselint, https://github.com/testthedocs/PlainLanguage/archive/refs/heads/master.zip

[...]

BasedOnStyles = Vale, Readability, alex, proselint, PlainLanguage
```

([source](https://github.com/wesley-dean-flexion/jubilant-computing-machine/blob/0da459143e076a828f03e42abd73379ebcb8e73a/vale.ini))

Running `vale sync` resulted in the following:

```
$ vale sync
 SUCCESS  Downloaded package 'Readability'
 SUCCESS  Downloaded package 'alex'
 SUCCESS  Downloaded package 'proselint'
Downloading packages [3/4] ██(shortened for brevity)███                                             75% | 2s
lstat /tmp/master839703475/master: no such file or directory
```

I initially thought this may have been a permission issue (the .zip file not extracting properly); however, even opening up the permissions on `/tmp` (which ought to be writeable by anyone to begin with) via `chmod 777 /tmp` had no effect.

I also verified that the URL to the .zip file worked properly

Attempting to run it without the downloaded package results in an as-expected failure on account of not having the style:

```
$ vale ./README.md 
E100 [loadStyles] Runtime error

style 'PlainLanguage' does not exist on StylesPath

Execution stopped with code 1.
```

(note: the `vale` above is running in a Docker container)

The .zip file was automatically provided by GitHub from the repo's master branch -- there's not a whole lot of configurability, as one could imagine.  A consequence of this is that the directory created by unzipping this file is that files are extracted to `PlainLanguage-master/` .

I suspect that if the file unzipped properly, one may be able to adjust the `BasedOnStyles` line to reference `PlainLanguage-master` (instead of `PlainLanguage`).  However, as the file isn't downloading and/or extracting properly, that's less helpful at the moment.

I was able to confirm that the URL provided by GitHub downloads and extracts properly:

```
$ wget -qO- 'https://github.com/testthedocs/PlainLanguage/archive/refs/heads/master.zip' | busybox unzip -
Archive:  -
  creating: PlainLanguage-master/
  inflating: PlainLanguage-master/ComplexWords.yml
  inflating: PlainLanguage-master/Contractions.yml
  inflating: PlainLanguage-master/LICENSE
  inflating: PlainLanguage-master/PassiveVoice.yml
  inflating: PlainLanguage-master/README.md
  inflating: PlainLanguage-master/SentenceLength.yml
  inflating: PlainLanguage-master/Slash.yml
  inflating: PlainLanguage-master/Wordiness.yml
  inflating: PlainLanguage-master/Words.yml
```

I was able to work around this by using the `jdkato/vale` image as a base, downloading the file, and creating a new .zip file that Vale is able to access properly:

```sh
  mkdir -p /PlainLanguage \
  && wget https://github.com/testthedocs/PlainLanguage/archive/refs/heads/master.zip -O /PlainLanguage/master.zip \
  && unzip -j -d /PlainLanguage /PlainLanguage/master.zip \
  && rm -f /PlainLanguage/master.zip \
  && zip /PlainLanguage.zip /PlainLanguage \
```

([source](https://github.com/wesley-dean-flexion/jubilant-computing-machine/blob/2455d849854771b2630c825da7db60ba17c25791/Dockerfile))

It might be nice if a package source could be expressed as a URL to a .git file, something like if the package matches `\.git$` then `git clone` the URL to the `StylesPath`

**Comment by jdkato** [comment]
`.zip` files (*[packages](https://vale.sh/docs/topics/packages/)*) have a required structure.

I have no plans to support Git (or any other VCS). You can find examples of how to "package" Git repos at https://vale.sh/hub/ (you only zip the config file and/or the `StylesPath`, not the entire repo).
